### DISCUSSION TITLE
updating symfony version using composer

### DISCUSSION THREAD

**Discussion by juantruquillo** [symfony/symfony#46166] [Q&A]
Hello everyone, first of all I apologise for my writting, am not english native. This is the first time that i use community to try get some help about a proccess that am trying to do and after a long research of information on web. Am working on a project made with symfony v 3.3.* and I am looking to upgrade it to v5.4, when i run composer update in console and after changing my composer.json ("symfony/symfony": "3.3.*", >>>> "symfony/symfony": "3.4.*",), it starts to upgrade packages but when it gets to the part of incenteev bundle, a request for missing parameter awaits to be entered  (google map api key). I used the key that is already at config.yml but console doesn't show any message of success or failure. At the moment that i launch server again, an error is showed up saying "Notice: Undefined index: api_key".  What should I do since I've already tried to add new api key from google cloud? Which files do i need to reconfigure with api key to make the update proccess to go smoothly?. Any way to avoid this missing parameters making them autoloadable?.

- SS with composer Json data
![composerjson](https://user-images.githubusercontent.com/44101562/165120388-0d528368-3197-4fb0-8db9-022c98b1fa0c.PNG)

- SS with update proccess
![composerupdate](https://user-images.githubusercontent.com/44101562/165120375-b21dac1f-722d-4072-99f3-0d7718e60097.PNG)

- SS with key request
![apikeyrequest](https://user-images.githubusercontent.com/44101562/165120385-cff29b9c-3aab-419d-a0e7-3551e2a29df4.PNG)

**Comment by stof** [comment]
Well, without the error location about where this notice comes from, there is nothing we can do to help you.

**Reply by juantruquillo** [reply]
Thanks a lot for fast reply, due to company policies, am not able to openly share info, but here are 2 ss, one with browser result and the other with related issue file.

![ohgoogle](https://user-images.githubusercontent.com/44101562/165134327-d6cf1a04-f5bd-4ea4-bee6-8d1e4b1702f3.PNG)

![undefinedkey](https://user-images.githubusercontent.com/44101562/165134330-ddb52eeb-e7f5-4454-838c-b1d4698aa93d.PNG)

**Reply by juantruquillo** [reply]
need to clarify something, when am requested to enter api key at composer update proccess, sometimes shows an error saying that a string argument was given and system expects an array, if I go ["api_key_alphanumeric"] or ['api_key_alphanumeric'], no success or failure message, it directly jumps to new line to enter any command.

**Reply by fd6130** [reply]
What if you remove the " incenteev bundle"? Maybe something not compatible there after you have updated your symfony version. And also check at your config file, is it compatible?

**Reply by stof** [reply]
The incenteev/composer-parameter-handler is only asking you interactively to provide values for the parameters defined in your `.dist` file. If the content of the `.dist` file does not match what your project expects, that's a bug in your code, not in that handler.

**Reply by juantruquillo** [reply]
Sorry for such a late reply, I had a bike accident but am back to work now

**Reply by juantruquillo** [reply]
> What if you remove the " incenteev bundle"? Maybe something not compatible there after you have updated your symfony version. And also check at your config file, is it compatible?

If i directly remove the bundle and run composer update i get this:

![Captura](https://user-images.githubusercontent.com/44101562/167173703-dc6333df-8c51-4ebc-8ff1-7da0103fd68c.PNG)

Sorry, I don't know what to answer to your question about config.yml compatibility, there is nothing related to incenteev bundle at config file.

![Captura1](https://user-images.githubusercontent.com/44101562/167173916-54045c16-2b95-470a-b647-8dbd326be006.PNG)

**Reply by juantruquillo** [reply]
> The incenteev/composer-parameter-handler is only asking you interactively to provide values for the parameters defined in your `.dist` file. If the content of the `.dist` file does not match what your project expects, that's a bug in your code, not in that handler.

in all files where i search "api_key_google_maps" there is any of 2 different things: the api key value ( AIzaSyDSGCC9_.............) or the call for the parameter like "%api_key_google_maps%". But in parameters.yml.dist its called google_maps_api_key with null value and at parameters.yml there is no api key at all.

config.yml

![Captura4](https://user-images.githubusercontent.com/44101562/167207856-28f4521f-08d5-4749-bb30-a1232c3b294c.PNG)

![Captura5](https://user-images.githubusercontent.com/44101562/167207949-1b194743-8d51-4e3a-98aa-f59f5b109d32.PNG)

parameters.yml.dist
![Captura2](https://user-images.githubusercontent.com/44101562/167205715-b3da215b-17db-4fbd-bc79-fae2e26c9baf.PNG)

parameters.yml
![Captura3](https://user-images.githubusercontent.com/44101562/167205717-651ea077-705d-4e6d-8431-6453617f495e.PNG)

No matter if I modify any of those files and make'em with same api_key variable and value, at the end with composer update it request me to interactively add the argument for api_key and throw me the failure message about array vs string argument like the 3rd img that i I shared in this discussion (SS with key request).

new parameters.yml
![Captura7](https://user-images.githubusercontent.com/44101562/167208204-d40bee26-7e0c-465b-a487-49e419bb982d.PNG)


new parameters.yml.dist
![Captura8](https://user-images.githubusercontent.com/44101562/167208206-d9620d19-839f-46c5-a84c-2eaa0a9f487c.PNG)


failure message
![image](https://user-images.githubusercontent.com/44101562/167206767-983a39aa-a83a-42df-a76b-e4168f3ecfe9.png)

**Reply by stof** [reply]
Well, if your parameters.yml.dist does not contain the right parameter name according to what the rest of your project uses, the solution is simple: fix your parameters.yml.dist file to use the expected name.
